src/lib/Microsoft.Health.Events/EventCheckpointing/StorageCheckpointClient.cs

```diff
@@ -33,21 +34,30 @@ public StorageCheckpointClient(StorageCheckpointOptions options, ITelemetryLogge
         {
             EnsureArg.IsNotNull(options);
             EnsureArg.IsNotNullOrWhiteSpace(options.BlobPrefix);
-            EnsureArg.IsNotNullOrWhiteSpace(options.BlobStorageConnectionString);
-            EnsureArg.IsNotNullOrWhiteSpace(options.BlobContainerName);
             EnsureArg.IsNotNullOrWhiteSpace(options.CheckpointBatchCount);
 
+            var accountName = EnsureArg.IsNotNullOrWhiteSpace(options.BlobStorageAccountName);
+            var containerName = EnsureArg.IsNotNullOrWhiteSpace(options.BlobContainerName);
+
             BlobPrefix = options.BlobPrefix;
 
             _lastCheckpointMaxCount = int.Parse(options.CheckpointBatchCount);
             _checkpoints = new ConcurrentDictionary<string, Checkpoint>();
             _lastCheckpointTracker = new ConcurrentDictionary<string, int>();
-            _storageClient = new BlobContainerClient(options.BlobStorageConnectionString, options.BlobContainerName);
+
+            var checkpointUri = new Uri($"https://{accountName}.blob.core.windows.net/{containerName}");
```

Comment:
I think the options should contain the fill Uri instead of constructing it here. The reason is, the Uri may be different for Government and Sovereign regions.

Reply:
As another option you could accept the full Url to the blob storage endpoint as well as the container name. Then join the two together to build the final checkpointUri. That would get around the issue of different Uri's for different Government/Sovereign regions and also allow you to connect to build uri's to multiple containers if need be.

Reply:
There should be an option to build this from the resource id.  That should be less error prone then using the URI as Rob and Nate have mentioned which is problematic when targeting different clouds.

---
In reply to: [566303793](https://github.com/microsoft/iomt-fhir/pull/87#discussion_r566303793) [](ancestors = 566303793)

Reply:
There is BlobURIBuilder.  Unfortunately it doesn't appear to support different clouds built in but it would be better to use than string interpolation itself. [https://docs.microsoft.com/en-us/dotnet/api/azure.storage.blobs.bloburibuilder?view=azure-dotnet](https://docs.microsoft.com/en-us/dotnet/api/azure.storage.blobs.bloburibuilder?view=azure-dotnet)

---
In reply to: [566498449](https://github.com/microsoft/iomt-fhir/pull/87#discussion_r566498449) [](ancestors = 566498449,566303793)

Reply:
Added full url for now.